src/admin/resolvers/mutations/ScheduledItem.integration.ts

```diff
@@ -217,6 +225,116 @@ describe('mutations: ScheduledItem', () => {
         await eventTracker.getCall(0).args[0].scheduledCorpusItem.externalId
       ).to.equal(scheduledItem.externalId);
     });
+
+    it('should fail if user has read-only access', async () => {
+      const headers = {
+        name: 'Test User',
+        username: 'test.user@test.com',
+        groups: `group1,group2,no-access-for-you`,
+      };
+
+      const server = getServerWithMockedHeaders(headers);
+      await server.start();
+
+      const approvedItem = await createApprovedItemHelper(db, {
+        title: 'A test story',
+      });
+
+      const input: CreateScheduledItemInput = {
+        approvedItemExternalId: approvedItem.externalId,
+        scheduledSurfaceGuid: 'NEW_TAB_EN_US',
+        scheduledDate: '2100-01-01',
+      };
+
+      const result = await server.executeOperation({
+        query: CREATE_SCHEDULED_ITEM,
+        variables: { data: input },
+      });
+
+      // ...without success. There is no data
+      expect(result.data).to.be.null;
+
+      expect(result.errors).not.to.be.null;
+
+      if (result.errors) {
+        // And there is an access denied error
+        expect(result.errors[0].message).to.contain(ACCESS_DENIED_ERROR);
+      }
+
+      await server.stop();
+    });
+
+    it("should fail if user doesn't have access to specified scheduled surface", async () => {
+      const headers = {
+        name: 'Test User',
+        username: 'test.user@test.com',
+        groups: `group1,group2,${MozillaAccessGroup.NEW_TAB_CURATOR_DEDE}`,
+      };
+
+      const server = getServerWithMockedHeaders(headers);
+      await server.start();
+
+      const approvedItem = await createApprovedItemHelper(db, {
+        title: 'A test story',
+      });
+
+      const input: CreateScheduledItemInput = {
+        approvedItemExternalId: approvedItem.externalId,
+        scheduledSurfaceGuid: 'NEW_TAB_EN_US',
+        scheduledDate: '2100-01-01',
+      };
+
+      const result = await server.executeOperation({
+        query: CREATE_SCHEDULED_ITEM,
+        variables: { data: input },
+      });
+
+      // ...without success. There is no data
+      expect(result.data).to.be.null;
+
+      expect(result.errors).not.to.be.null;
+
+      if (result.errors) {
+        // And there is an access denied error
+        expect(result.errors[0].message).to.contain(ACCESS_DENIED_ERROR);
+      }
+
```

Comment:
I just learned from Kelvin the other day that you have to do this to be able to do optional chaining on an possibly undefined array 🙃 :

`result.errors?.[0].message` have to add the `.` before we can access the array index, which is absolutely not how you access array indices. With this you can get rid of the `if()`

Reply:
Thanks for the tip, @Herraj! There were quite a few instances of the `if()` statement being used around these checks in integration tests so I went and updated them all.